### DISCUSSION TITLE
Strict site isolation

### DISCUSSION THREAD

**Discussion by rsssxd** [bromite/bromite#2001] [Q&A]
Why is the following flag feature turned off?

![Screenshot_20220419-230403_Bromite](https://user-images.githubusercontent.com/60405617/164086771-41662c3f-43f6-4514-8c6b-1d00616302da.png)

**Comment by csagan5** [comment]
It is enabled via preferences; @uazo any benefit at not using the flag for this one?

**Reply by uazo** [reply]
see https://github.com/bromite/bromite/pull/1832#issuecomment-1051767668 and https://github.com/bromite/bromite/pull/1832#issuecomment-1063829936

**Reply by csagan5** [reply]
Still not clear. [This patch](https://github.com/bromite/bromite/blob/100.0.4896.135/build/patches/Enable-StrictOriginIsolation-and-SitePerProcess.patch) enables `StrictOriginIsolation` (via feature) and `kSitePerProcess` (via preference). The flag `#enable-site-per-process` would also override the `kSitePerProcess` preference and be equivalent: any reason why it is not used?

> `Strict Site Isolation` is superseded by `Strict Origin Isolation`, see my comment above "kStrictOriginIsolation is better than kStrictSiteIsolation"

If this is true then we do not need any change at all for `Strict Site Isolation`?

**Reply by uazo** [reply]
sorry I don't understand the question
code needs those changes to activate the function: do you want me to go back to the lines of the code and prove it to you?

**Reply by csagan5** [reply]
No; the questions are two:
1. what does it mean that `Strict Site Isolation` supersedes `Strict Origin Isolation`? If both changes are still needed in which way does one supersede the other?
2. would changing the default flag value for `#enable-site-per-process` be sufficient? (without the preference changes)

**Reply by uazo** [reply]
> what does it mean that Strict Site Isolation supersedes Strict Origin Isolation?

however it is the other way around: `Strict Origin Isolation` supersedes `Strict Site Isolation`.
"supersedes" in the sense that you either have one or the other active option.

> would changing the default flag value for #enable-site-per-process be sufficient?

No, because it is a [switch](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/about_flags.cc;l=3947;drc=5e62802ab420dc1743a1824ec4b537a730b2b24b;bpv=1;bpt=1?q=enable-site-per-process&ss=chromium%2Fchromium%2Fsrc), not a flag

**Reply by csagan5** [reply]
> > what does it mean that Strict Site Isolation supersedes Strict Origin Isolation?
> 
> however it is the other way around: `Strict Origin Isolation` supersedes `Strict Site Isolation`. "supersedes" in the sense that you either have one or the other active option.

Why do we have both enabled?

> > would changing the default flag value for #enable-site-per-process be sufficient?
> 
> No, because it is a [switch](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/about_flags.cc;l=3947;drc=5e62802ab420dc1743a1824ec4b537a730b2b24b;bpv=1;bpt=1?q=enable-site-per-process&ss=chromium%2Fchromium%2Fsrc), not a flag

It is a flag linked to a switch; it would need a feature to work as regular flags, if I understand correctly. I will remove it since it does nothing.

**Reply by uazo** [reply]
> it would need a feature to work as regular flags

correct, unless it is due to bugs introduced by other patches

**Reply by uazo** [reply]
> Why do we have both enabled?

it wouldn't make sense because they affect how the renderprocess is started

**Reply by csagan5** [reply]
> > Why do we have both enabled?
> 
> it wouldn't make sense because they affect how the renderprocess is started

We currently have both site-per-process isolation and strict site isolation; if strict origin isolation replaces strict site isolation then we should remove it because redundant.

**Reply by sunilsingh083proton** [reply]
> It is enabled via preferences; @uazo any benefit at not using the flag for this one?

Dear Developer, in the latest version of Bromite the flag "strict site isolation" is not present, i asked other users of Bromite, all others told me the same. Its not visible in the flags, whereas in Chrome its visibly there.

**Reply by sunilsingh083proton** [reply]
> > > what does it mean that Strict Site Isolation supersedes Strict Origin Isolation?
> > 
> > 
> > however it is the other way around: `Strict Origin Isolation` supersedes `Strict Site Isolation`. "supersedes" in the sense that you either have one or the other active option.
> 
> Why do we have both enabled?
> 
> > > would changing the default flag value for #enable-site-per-process be sufficient?
> > 
> > 
> > No, because it is a [switch](https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/about_flags.cc;l=3947;drc=5e62802ab420dc1743a1824ec4b537a730b2b24b;bpv=1;bpt=1?q=enable-site-per-process&ss=chromium%2Fchromium%2Fsrc), not a flag
> 
> It is a flag linked to a switch; it would need a feature to work as regular flags, if I understand correctly. I will remove it since it does nothing.


Dear Developer, in the latest version of Bromite the flag "strict site isolation" is not present, i asked other users of Bromite, all others told me the same. Its not visible in the flags, whereas in Chrome its visibly there. I tried to contact to XDA as well.
Please Check.

Regards

**Reply by csagan5** [reply]
There is no problem, it is already enabled by a patch.

**Reply by sunilsingh083proton** [reply]
Dear Developer,
I will like to inform you that the "Strict origin isolation" Flag is present but the other Flag " Strict Site Isolation" is NOT visible at all in the flags in the latest version. 

I have cross checked with other Bromite users and they have all confirmed the same. Whereas Brave & Chrome both of these browsers have both the Flags visibly present.

Therefore, I would humbly request you to kindly have a recheck of the Flags in the latest version and you will be surprised.

Regards

**Comment by sunilsingh083proton** [comment]
> Why is the following flag feature turned off?
> 
> ![Screenshot_20220419-230403_Bromite](https://user-images.githubusercontent.com/60405617/164086771-41662c3f-43f6-4514-8c6b-1d00616302da.png)


In the latest version of Bromite the flag "strict site isolation" is not present, i have asked other users of Bromite, all others told me the same. Its not visible in the flags, whereas in Chrome its visibly there.
